### DISCUSSION TITLE
Multiple stages for ASE interface

### DISCUSSION THREAD

**Discussion by enssow** [Tomoki-YAMASHITA/CrySPY#55] [Q&A] [answered]
Hi! 
I've been using the CrySPY package over the last week, understanding different features etc. and it's great :) But I'm having issues with setting up more than one stage (i.e. nstage >1) when using an ASE algorithm implemented in python (i.e. ase_in.py). After setting up the same number of python files as stages I still get an error saying: 
```
File "/root/env/lib64/python3.9/site-packages/cryspy/interface/select_code.py", line 75, in next_stage
    skip_flag = ctrl_job_ase.next_stage_ase(stage, work_path)
  File "/root/env/lib64/python3.9/site-packages/cryspy/interface/ASE/ctrl_job_ase.py", line 36, in next_stage_ase
    shutil.copyfile(fincar, work_path+'INCAR')
  File "/usr/lib64/python3.9/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: './calc_in/INCAR_2'
```
The problem seems to arise in the function `next_stage_ase` in `src/cryspy/interface/ASE/ctrl_job_ase.py` which tries to copy an INCAR file but these aren't required if the structure is optimised by ASE. However, if they are, would you be able to advise me on how to work around this? 
Thank you!

**Comment by Tomoki-YAMASHITA** [comment, marked as answer]
This is just a bug. I have just uploaded the modified version (1.2.1).
You can upgrade by `pip install -U csp-cryspy`.

Thank you for reporting.
